src/main/groovy/io/openliberty/tools/gradle/utils/LooseEarApplication.groovy

```diff
@@ -1,42 +1,50 @@
+import org.w3c.dom.Element
 
 public class LooseEarApplication extends LooseApplication {
     
     protected Task task;
+    protected Logger logger;
 
-    public LooseEarApplication(Task task, LooseConfigData config) {
-        super(task.getProject().getBuildDir().getAbsolutePath(), config)
+    public LooseEarApplication(Task task, LooseConfigData config, Logger logger) {
+        super(task.getProject().getLayout().getBuildDirectory().getAsFile().get().getAbsolutePath(), config)
         this.task = task
+        this.logger = logger
     }
 
     public void addSourceDir() throws Exception {
-        File sourceDir = new File(task.getProject().path.replace(":","") + "/" + task.getProject().getConvention().getPlugin(EarPluginConvention.class).getAppDirName())
-        config.addDir(sourceDir, "/")
+        if (task.getProject().getPlugins().hasPlugin("ear")) {
+            Ear ear = (Ear) task.getProject().ear
+            File sourceDir = new File(task.getProject().path.replace(":","") + "/" + ear.getAppDirectory().getAsFile().get().getAbsolutePath())
+            config.addDir(sourceDir, "/")
+        }
     }
 
     public void addApplicationXmlFile() throws Exception {
-        String applicationName = "/" + task.getProject().getConvention().getPlugin(EarPluginConvention.class).getDeploymentDescriptor().getFileName()
-        File applicationXmlFile = new File(task.getProject().path.replace(":","") + "/" + task.getProject().getConvention().getPlugin(EarPluginConvention.class).getAppDirName() + "/META-INF/" + applicationName)
-        if (applicationXmlFile.exists()) {
-            config.addFile(applicationXmlFile, "/META-INF/application.xml");
+        if (task.getProject().getPlugins().hasPlugin("ear")) {
+            Ear ear = (Ear) task.getProject().ear
+            String applicationName = "/application.xml"
+            if (ear.getDeploymentDescriptor() != null){
+                applicationName = "/" + ear.getDeploymentDescriptor().getFileName()
+            }
+            File applicationXmlFile = new File(task.getProject().path.replace(":", "") + "/" + ear.getAppDirectory().getAsFile().get().getAbsolutePath() + "/META-INF/" + applicationName)
+            if (applicationXmlFile.exists()) {
+                config.addFile(applicationXmlFile, "/META-INF/application.xml");
+            } else {
+                applicationXmlFile = new File(task.getDestinationDirectory().get().getAsFile().getParentFile().getAbsolutePath() + "/tmp/ear" + applicationName);
+                config.addFile(applicationXmlFile, "/META-INF/application.xml");
+            }
         }
         else {
-            applicationXmlFile = new File(task.getDestinationDirectory().get().getAsFile().getParentFile().getAbsolutePath() + "/tmp/ear" + applicationName);
-            config.addFile(applicationXmlFile, "/META-INF/application.xml");
+            logger.warn("Project is expected to have EAR plugin. Application xml file may not be added correctly")
```

Comment:
"Could not get the application.xml file location from the EAR plugin because it is not configured. The file may not be added correctly to the application archive."

Reply:
Done